include/drivers/can.h

```diff
@@ -93,25 +93,28 @@ extern "C" {
  * `CAN_NO_FREE_FILTER` is returned by `can_attach_*()` if no free filters are
  * available. `CAN_TIMEOUT` indicates that @a can_recover() timed out.
  *
+ * @warning These definitions are deprecated. Use the corresponding errno
+ * definitions instead.
+ *
  * @{
  */
 
 /** Transmitted successfully. */
-#define CAN_TX_OK          (0)
+#define CAN_TX_OK          __DEPRECATED_MACRO (0)
 /** General transmit error. */
-#define CAN_TX_ERR         (-2)
+#define CAN_TX_ERR         __DEPRECATED_MACRO (-EIO)
 /** Bus arbitration lost during transmission. */
-#define CAN_TX_ARB_LOST    (-3)
+#define CAN_TX_ARB_LOST    __DEPRECATED_MACRO (-EBUSY)
 /** CAN controller is in bus off state. */
-#define CAN_TX_BUS_OFF     (-4)
+#define CAN_TX_BUS_OFF     __DEPRECATED_MACRO (-ENETDOWN)
 /** Unknown error. */
-#define CAN_TX_UNKNOWN     (-5)
+#define CAN_TX_UNKNOWN     __DEPRECATED_MACRO (CAN_TX_ERR)
 /** Invalid parameter. */
-#define CAN_TX_EINVAL      (-22)
+#define CAN_TX_EINVAL      __DEPRECATED_MACRO (-EINVAL)
 /** No free filters available. */
-#define CAN_NO_FREE_FILTER (-1)
+#define CAN_NO_FREE_FILTER __DEPRECATED_MACRO (-ENOSPC)
 /** Operation timed out. */
-#define CAN_TIMEOUT        (-1)
+#define CAN_TIMEOUT        __DEPRECATED_MACRO (-EAGAIN)
```

Comment:
note: `ETIMEDOUT` may be used?

Reply:
Yes. It was either that or `-EAGAIN`. I settled on the latter since this seems to be dominant in existing Zephyr APIs for reporting a timeout on a retryable operation.